pinot-plugins/pinot-stream-ingestion/pinot-pulsar/src/main/java/org/apache/pinot/plugin/stream/pulsar/PulsarStreamMetadataProvider.java

```diff
@@ -124,19 +127,28 @@ public StreamPartitionMsgOffset fetchStreamPartitionOffset(OffsetCriteria offset
 
         for (int p = newPartitionStartIndex; p < partitionedTopicNameList.size(); p++) {
 
-          Reader reader =
-              _pulsarClient.newReader().topic(getPartitionedTopicName(p)).startMessageId(_config.getInitialMessageId())
-                  .create();
+          Consumer consumer = _pulsarClient.newConsumer().topic(getPartitionedTopicName(p))
```

Comment:
getPartitionedTopicName is making a call to `_pulsarClient.getPartitionsForTopic(_topic).get()` for every partition. Can we call it just once before the loop?